peerprep-fe/src/app/(main)/components/problems/ProblemTable.tsx

```diff
@@ -1,38 +1,53 @@
 'use client';
-import { useSearchStore } from '@/state/useSearchStore';
-import { useMemo } from 'react';
 import { Problem } from '@/types/types';
 import ProblemRow from './ProblemRow';
+import { Skeleton } from '@/components/ui/skeleton';
 
 interface ProblemTableProps {
   problems: Problem[];
+  isLoading: boolean;
 }
 
-export default function ProblemTable({ problems }: ProblemTableProps) {
-  const { searchTerm } = useSearchStore();
-
-  const filteredProblems = useMemo(() => {
-    return problems.filter((problem) =>
-      problem.title.toLowerCase().includes(searchTerm.toLowerCase()),
-    );
-  }, [problems, searchTerm]);
-
+export default function ProblemTable({
+  problems,
+  isLoading,
+}: ProblemTableProps) {
   return (
     <div className="overflow-x-auto">
       <table className="w-full border-collapse">
         <thead>
           <tr className="border-b border-gray-700 text-left">
-            <th className="px-4 py-2">Status</th>
-            <th className="px-4 py-2">Title</th>
+            <th className="w-28 px-4 py-2">Status</th>
+            <th className="w-1/3 px-4 py-2">Title</th>
             <th className="px-4 py-2">Topics</th>
             <th className="px-4 py-2">Difficulty</th>
           </tr>
         </thead>
         <tbody>
-          {filteredProblems.map((problem) => (
-            // should not use index as key
-            <ProblemRow key={problem._id} problem={problem} />
-          ))}
+          {isLoading
```

Comment:
Not sure why when i refresh, it shows the sign up page then skeleton then actual questions HAHAHHA we might need to make a new issue to fix that